Packs/GenericSQL/Integrations/GenericSQL/GenericSQL_test.py

```diff
@@ -10,6 +10,32 @@
 from GenericSQL import Client, sql_query_execute, generate_default_port_by_dialect
 
 
+class MockedClient(Client):
+    """
+    This class presents a client object for testing the cache engines mechanism.
+    """
+
+    def __init__(self, mocker, global_cache: dict, cache_string: str, dialect: str, host: str, username: str, password: str,
+                 port: str, database: str, connect_parameters: str, ssl_connect: bool, use_pool: bool):
+
+        mocker.patch.object(Client, '_convert_dialect_to_module')
+        mocker.patch.object(Client, '_generate_db_url')
+        mocker.patch.object(Client, '_get_global_cache', return_value=global_cache)
+        mocker.patch.object(Client, '_get_cache_string', return_value=cache_string)
+        super().__init__(dialect, host, username, password, port, database, connect_parameters, ssl_connect, use_pool)
```

Comment:
```suggestion
        def _convert_dialect_to_module:
            ...            
```

and the same for all others, overriding them

Reply:
Done